xml/ntp_banners.xml

```diff
@@ -419,50 +419,382 @@ NTP "banners", taken from a readvar response
+    <param pos="0" name="service.product" value="NTP"/>
+    <param pos="1" name="service.version"/>
+    <param pos="2" name="os.arch"/>
+    <param pos="3" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^.*version=&quot;ntpd ([^ ]+)[^&quot;]+&quot;,.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/?5.([789]|10)&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+    <description>ntpd running on Solaris 7-Solaris 10 (SunOS/5.7 - SunOS/5.10) </description>
+    <example service.version="4.2.0@1.1161-r" os.arch="sun4u" os.version="7">
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS/5.7", leap=00, stratum=2,
+    </example>
+    <example service.version="4.2.0@1.1161-r" os.arch="sun4u" os.version="10">
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS/5.10", leap=00, stratum=2,
+    </example>
+    <example service.version="4.2.0@1.1161-r" os.arch="sun4u" os.version="7">
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS5.7", leap=00,stratum=2,
+    </example>
+    <example service.version="4.2.0@1.1161-r" os.arch="sun4u" os.version="10">
+      version="ntpd 4.2.0@1.1161-r Wed Apr 20 11:28:05 EDT 2005 (1)",
+      processor="sun4u", system="SunOS5.10", leap=00, stratum=2,
+    </example>
+    <param pos="0" name="os.vendor" value="Sun"/>
+    <param pos="0" name="os.family" value="Solaris"/>
+    <param pos="0" name="os.product" value="Solaris"/>
+    <param pos="0" name="service.family" value="NTP"/>
+    <param pos="0" name="service.product" value="NTP"/>
+    <param pos="1" name="service.version"/>
+    <param pos="2" name="os.arch"/>
+    <param pos="3" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="^.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/?5.0&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+   <description>Solaris 2.0 (SunOS/5.0) with no ntp version</description>
+   <example os.arch="sun4m">
+     processor="sun4m", system="SunOS/5.0"
+   </example>
+   <example os.arch="sun4u">
+    processor="sun4u", system="SunOS/5.0"
+   </example>
+   <example os.arch="sun4m">
+     processor="sun4m", system="SunOS5.0"
+   </example>
+   <example os.arch="sun4u">
+     processor="sun4u", system="SunOS5.0"
+   </example>
+   <param pos="0" name="os.vendor" value="Sun"/>
+   <param pos="0" name="os.family" value="Solaris"/>
+   <param pos="0" name="os.product" value="Solaris"/>
+   <param pos="0" name="os.version" value="2.0" />
+   <param pos="1" name="os.arch"/>
+  </fingerprint>
+  <fingerprint pattern="^.*processor=&quot;([^ ]+)&quot;,.*system=&quot;SunOS/?5.1&quot;" flags="REG_DOT_NEWLINE,REG_ICASE">
+   <description>Solaris 2.1 (SunOS/5.1) with no ntp version</description>
+   <example os.arch="sun4m">
+     processor="sun4m", system="SunOS/5.1"
+   </example>
+   <example os.arch="sun4u">
```

Comment:
Really minor, but when adding examples, in particular those in which you are testing the extracted values, I prefer to only test banner/other data that is a distinct use-case covered by the regex in question.  For example, here and throughout this PR, there are cases where you are ensuring that a regex like `([^ ]+)` properly captures non-whitespace properly.  If it captures `sun4u`, it will catch `sun4m`.  

IOW, the purpose of the examples and their tests is not to list all banners we've seen that we expect to match, but should be the unique subset of those that specifically test the particulars of the regex.

The reason this is a little important is that every added example increases the time that the tests take to run, in particular because *every* example is tested against *every* regex (in the same file) that came before it to ensure that the example matches the correct regex, since order is important in these files.  So, for this single fingerprint where you've added 3 somewhat duplicate examples, each of these must be tested against the dozens of `pattern`s that came before it.  Over time this makes the tests take longer and longer without providing additional coverage.

Contrast this with a regex that did something like `(\d+|alpha|beta)`, for capturing a version field that might have digits in it or might simply have "alpha" or "beta".  In cases like this, IMO it would be appropriate to add examples for each of these 3 cases, since those are 3 distinct things that the regex is designed to capture.